src/components/popover-container/popover-container.style.ts

```diff
@@ -32,29 +57,8 @@ const PopoverContainerContentStyle = styled.div<PopoverContainerContentStyleProp
   position: absolute;
   z-index: ${({ theme }) => theme.zIndex.popover};
 
-  ${({ animationState }) => {
-    switch (animationState) {
-      case "entering":
-        return `
-        opacity: 0;
-        transform: translateY(-8px);
-      `;
-      case "entered":
-        return `
-        opacity: 1; 
-        transform: translateY(0);
-        transition: all 0.3s cubic-bezier(0.25, 0.25, 0, 1.5);
-          `;
-      case "exiting":
-        return `
-        opacity: 0; 
-        transform: translateY(-8px);
-        transition: all 0.3s cubic-bezier(0.25, 0.25, 0, 1.5);
-          `;
-      default:
-        return "opacity: 0";
-    }
-  }}
+  ${({ animationState, disableAnimation }) =>
+    disableAnimation ? `opacity: 1` : animationToRender(animationState)}
```

Comment:
suggestion (non-blocking): you could remove the need for a ternary here and push it to the `animationToRender`

```
function animationToRender(animationState?: TransitionStatus, disableAnimation?: boolean) {
  switch (animationState) {
    case "entering":
      return `
        opacity: 0;
        transform: translateY(-8px);
      `;
    case "entered":
      return `
        opacity: 1; 
        transform: translateY(0);
        transition: all 0.3s cubic-bezier(0.25, 0.25, 0, 1.5);
      `;
    case "exiting":
      return `
        opacity: 0; 
        transform: translateY(-8px);
        transition: all 0.3s cubic-bezier(0.25, 0.25, 0, 1.5);
      `;
    default:
      return `opacity: ${disableAnimation ? 1 : 0}`;
  }
}
...
  ${({ animationState, disableAnimation }) => animationToRender(animationState, disableAnimation)}